subprojects/model-core/src/test/groovy/org/gradle/internal/instantiation/generator/IdentityClassGenerator.java

```diff
@@ -97,7 +98,7 @@ public int getModifiers() {
                         }
                     });
                 }
-                return constructors;
+                return Ordering.from(new ConstructorComparator()).sortedCopy(constructors);
```

Comment:
It makes me sad that we shade and relocate Guava. I know it's a tall order to remove it, but do we have any other options here to not expand our usage of it?

Reply:
This is a test-only fixture, so I don't think it has quite the impact/proliferation problem.

The other place this is used is in: https://github.com/gradle/gradle/pull/18319/files?file-filters%5B%5D=.java#diff-643878892c15713ab48325c9ffc77733bff0dec4ee696e5d9c3f9cf0e4982b6bR440

Our class decorator/generator is already using other Guava things (e.g., TypeToken), so it's unlikely that we'd remove Guava from there.

But I do agree that we should be mindful of where Guava gets in (it can't be a part of a public API and it needs to stay away from entry points like the wrapper that need to be kept lean).